src/hf_hydrodata/point.py

```diff
@@ -1637,53 +1603,68 @@ def _get_sites(
         grid_bounds_sites = list(grid_bounds_df["site_id"])
 
         if len(grid_bounds_sites) > 0:
-            grid_bounds_query = """ AND s.site_id IN (%s)""" % ",".join(
+            grid_bounds_query = " AND s.site_id IN (%s)" % ",".join(
                 "?" * len(grid_bounds_sites)
             )
             for s in grid_bounds_sites:
                 param_list.append(s)
         else:
             raise Exception("There are no sites within the provided grid_bounds.")
     else:
-        grid_bounds_query = """"""
+        grid_bounds_query = ""
+
+    # HUC ID filter
+    if "huc_id" in options and options["huc_id"] is not None:
+        # Make sure that the option "grid" is defined
+        try:
+            assert "grid" in options and options["grid"] in ("conus1", "conus2")
+        except:
+            raise ValueError(
+                "When providing the parameter `huc_id`, please also provide the parameter `grid` as either 'conus1' or 'conus2'."
+            )
+        huc_query, param_list = _get_huc_query(
+            options, param_list, conn, dataset=dataset, variable=variable
+        )
+    else:
+        huc_query = ""
```

Comment:
Is there a way to make this a reusable function and collapse this with L744-L756? It could live in a utils directory 😜

Reply:
Good point :) I will think about a good way to do this - I may just expand the `_get_huc_query` utility that I'm sharing across these code segments to include some more of this logic.